internal/pkg/cli/task_run.go

```diff
@@ -49,6 +53,10 @@ const (
 	fmtImageURI = "%s:%s"
 )
 
+type cmdGenerator interface {
+	Generate() (*generator.GenerateCommandOpts, error)
+}
```

Comment:
This is interesting 💭 this is the first time that we have a package that populates values for a command. 

Did we go with this route to keep the `cli` pkg small? I think that makes sense to me, but I wonder if instead of a new pkg we could have moved most of the functionality into the `internal/pkg/ecs` pkg and removed the `generator` pkg altogether.   
This functionality seems like should belong in the`cli` pkg since it can only be used by `task_run` so there is no other possible consumer of the `generator` pkg:
https://github.com/aws/copilot-cli/blob/f2ad67677e3cef3a58e3bebe7dafa63adffeae00/internal/pkg/generator/generate.go#L74-L75

Hope that makes sense! Instead, we can maybe move this functionality to the `ecs` pkg: 
https://github.com/aws/copilot-cli/blob/f2ad67677e3cef3a58e3bebe7dafa63adffeae00/internal/pkg/generator/ecs_service.go#L29 to something like:
 ```go
 func RunTaskRequestFromECSService(client ecsDescriber, cluster, service string) (*RunTaskRequest, error)
 func RunTaskRequestFromService(client ecsDescriber, app, env, svc string) (*RunTaskRequest, error)
 ```